### PR TITLE
[Dependencies] Pin pip3 version

### PR DESCRIPTION
**Pull request by jleveque**
#### Why I did it

A new version of pip (21.1) was released on 4/24, and it appears to have introduced new behavior which is causing the build to break. New log messages such as the following are seen in the build log:

```
WARNING: Value for scheme.platlib does not match. Please report this to <https://github.com/pypa/pip/issues/9617> 
WARNING: Value for scheme.headers does not match. Please report this to <https://github.com/pypa/pip/issues/9617> 
WARNING: Value for scheme.scripts does not match. Please report this to <https://github.com/pypa/pip/issues/9617> 
WARNING: Value for scheme.data does not match. Please report this to <https://github.com/pypa/pip/issues/9617> 
```

And the build is failing with errors like:

```
Requirement already satisfied: sonic-yang-mgmt in /usr/local/lib/python3.7/dist-packages (from sonic-utilities==1.2) (1.0) 
Collecting pexpect==4.8.0 
Downloading pexpect-4.8.0-py2.py3-none-any.whl (59 kB) 
ERROR: Cannot install sonic-utilities==1.2 because these package versions have conflicting dependencies. 
The conflict is caused by: 
sonic-utilities 1.2 depends on netifaces==0.10.7 
The user requested (constraint) netifaces==0.10.9 
To fix this you could try to: 
1. loosen the range of package versions you've specified 
2. remove package versions to allow pip attempt to solve the dependency conflict 
```

Until we can resolve these issues, pin the version to the last known good version, 21.0.1 in Buster environments, and 20.3.4 in Stretch environments.

---

Update: This appears to be due to netifaces version 0.10.9 being [installed](https://github.com/Azure/sonic-buildimage/blob/master/platform/vs/docker-sonic-vs/Dockerfile.j2#L113) in docker-sonic-vs, whereas sonic-utilities explicitly requires 0.10.7 in its setup.py file. Older versions of pip did not error out here, but 21.1 does.

https://github.com/Azure/sonic-utilities/pull/1530 may solve this issue by relaxing the dependencies, therefore this PR may not be necessary.

### CODE REVIEW

**Comment by lguohan** [comment]
can you look into the docker-sonic-vs/Dockerfile.j2, the build breaks there.

**Comment by jleveque** [comment]
> can you look into the docker-sonic-vs/Dockerfile.j2, the build breaks there.

docker-sonic-vs uses docker-config-engine-buster as its parent, which in turn uses docker-base-buster as its parent, so this PR should cover it, also.

File: sonic-slave-buster/Dockerfile.j2 [hunk 1]
```diff
@@ -395,7 +395,7 @@ RUN export VERSION=1.14.2 \
  && echo 'export PATH=$PATH:$GOROOT/bin' >> /etc/bash.bashrc \
  && rm go$VERSION.linux-*.tar.gz
 
-RUN pip3 install --upgrade pip
+RUN pip3 install --upgrade 'pip==21.0.1'
```
Comment (joyas-joseph) [inline sonic-slave-buster/Dockerfile.j2:398]: With 21.0.1, sonic_yang_models build is broken.
```
----------------------------- Captured stdout call -----------------------------
Failed: pyang -f tree ./yang-models/*.yang > ./yang-models/sonic_yang_tree
----------------------------- Captured stderr call -----------------------------
./yang-models/sonic-acl.yang:8: error: module "ietf-inet-types" not found in search path
```
Can we use version 20.3.4 for buster also?

Comment (jleveque) [reply sonic-slave-buster/Dockerfile.j2:398]: Why do you ask? Can you please provide some reasoning?

Comment (joyas-joseph) [reply sonic-slave-buster/Dockerfile.j2:398]: Build failure is seen for sonic_yang_models package with pip versions higher than 20.3.4.

The error can be seen in https://github.com/Azure/sonic-buildimage/pull/7407 also.

Comment (jleveque) [reply sonic-slave-buster/Dockerfile.j2:398]: Have you verified that pip is the problem here? Why was this error not seen when using pip 21.0, which has been available since January?

Comment (joyas-joseph) [reply sonic-slave-buster/Dockerfile.j2:398]: Yes. I have verified that pip version is the issue here. The build goes through fine when using 20.3.4. With version 21, they made a change in pip that breaks pyang.

Comment (joyas-joseph) [reply sonic-slave-buster/Dockerfile.j2:398]: Let me try again with the latest to ensure the build really goes through

Comment (joyas-joseph) [reply sonic-slave-buster/Dockerfile.j2:398]: My bad! When verifying build for version 21.0.1, I still ended up with 21.1.

The breakage came with 21.1. We are good with 21.0.1.

Sorry.

Comment (jleveque) [reply sonic-slave-buster/Dockerfile.j2:398]: No worries! Good to know. Thanks for confirming!

**Comment by jleveque** [comment]
Closing as this appears to have been fixed by https://github.com/Azure/sonic-utilities/pull/1530
